runtime/vm/resolvesupport.cpp

```diff
@@ -1574,7 +1574,10 @@ resolveVirtualMethodRefInto(J9VMThread *vmStruct, J9ConstantPool *ramCP, UDATA c
 		if (method != NULL) {
 #if defined(J9VM_OPT_VALHALLA_NESTMATES)
 			J9ROMMethod* romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
-			if (J9_ARE_ALL_BITS_SET(romMethod->modifiers, J9AccPrivate)) {
+			/* Only allow non-interface method to call invokePrivate, private interface method should use "invokeInterface" bytecode
+			 * The else case will throw ICCE for private interface method 
+			 */
```

Comment:
is this a spec requirement?  Does this match the RIs behaviour?

Reply:
yes, this does match RIs behaviour